### PR TITLE
Add samsung partition information table

### PR DESCRIPTION
**Pull request by evdenis**

### CODE REVIEW

**Comment by evdenis** [comment]
Example file (gunzip first):
[BEYOND1LTE_EUR_OPEN.pit.gz](https://github.com/kaitai-io/kaitai_struct_formats/files/5958298/BEYOND1LTE_EUR_OPEN.pit.gz)

**Comment by evdenis** [comment]
Example usage:
 - https://cruelkernel.org/tools/pit/
 - https://github.com/CruelKernel/webpit

File: filesystem/samsung_pit.ksy [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  file-extension: pit
+  endian: le
+  imports:
+    - /common/samsung_signature
+
+doc: |
+  Reversed format of partition information table used in
+  samsung decices.
+
+doc-ref: https://gitlab.com/BenjaminDobell/Heimdall/-/tree/master/libpit/source
```
Comment (KOLANICH) [inline filesystem/samsung_pit.ksy:14]: https://github.com/kaitai-io/kaitai_struct_formats/pull/302/files#diff-41ce83ce4a8bc1c0b8eb954dc851ef7a877bc4c92b13391a407fe2ffcbab6d2eR9-R13

File: filesystem/samsung_pit.ksy [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        type: strz
+        size: 32
+        encoding: ASCII
+      - id: fota_filename
+        type: strz
+        size: 32
+        encoding: ASCII
+
+enums:
+  binary_type:
```
Comment (KOLANICH) [inline filesystem/samsung_pit.ksy:104]: https://github.com/kaitai-io/kaitai_struct_formats/pull/302/files#diff-41ce83ce4a8bc1c0b8eb954dc851ef7a877bc4c92b13391a407fe2ffcbab6d2eR100-R121

File: filesystem/samsung_pit.ksy [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    0: ap
+    1: cp
+  device_type:
```
Comment (KOLANICH) [inline filesystem/samsung_pit.ksy:107]: https://github.com/kaitai-io/kaitai_struct_formats/pull/302/files#diff-41ce83ce4a8bc1c0b8eb954dc851ef7a877bc4c92b13391a407fe2ffcbab6d2eR107-R121

File: filesystem/samsung_pit.ksy [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+      - id: value
+        type: u4
+    instances:
+      read:
+        value: true
+      write:
+        value: (value & 1) == 0x1
+      stl:
+        value: (value & (1 << 1)) == 0x2
+  update_attributes:
```
Comment (KOLANICH) [inline filesystem/samsung_pit.ksy:55]: https://github.com/kaitai-io/kaitai_struct_formats/pull/302/files#diff-41ce83ce4a8bc1c0b8eb954dc851ef7a877bc4c92b13391a407fe2ffcbab6d2eR85-R93

File: filesystem/samsung_pit.ksy [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  - id: partitions
+    type: partition
+    repeat: expr
+    repeat-expr: num_entries
+  - id: signature
+    type: samsung_signature
+    if: not _io.eof
+
+types:
+  attributes:
```
Comment (KOLANICH) [inline filesystem/samsung_pit.ksy:44]: https://github.com/kaitai-io/kaitai_struct_formats/pull/302/files#diff-41ce83ce4a8bc1c0b8eb954dc851ef7a877bc4c92b13391a407fe2ffcbab6d2eR68-R84

File: filesystem/samsung_pit.ksy [hunk 6]
```diff
@@ -0,0 +1,111 @@
+meta:
+  id: samsung_pit
```
Comment (KOLANICH) [inline filesystem/samsung_pit.ksy:2]: firmware/samsung_android_pit.ksy

File: common/samsung_signature.ksy [hunk 7]
```diff
@@ -0,0 +1,42 @@
+meta:
+  id: samsung_signature
```
Comment (KOLANICH) [inline common/samsung_signature.ksy:2]: security/samsung_android_signature

File: common/samsung_signature.ksy [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  title: Samsung Crypto Signature for Android Firmware Images
```
Comment (KOLANICH) [inline common/samsung_signature.ksy:3]: I don't see any crypto in this spec.

File: common/samsung_signature.ksy [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  license: CC0-1.0
+
+doc: |
+  Signature that samsung appends to android images files.
+  Only part of the fields reversed. A phone's bootloader
+  checks images signature validity during boot.
+
+types:
+  samsung_signature:
```
Comment (KOLANICH) [inline common/samsung_signature.ksy:12]: Should likely be a `_root` level type in this spec.

**Comment by evdenis** [comment]
Ok, seems like this a duplicate. I'll close the PR

File: common/samsung_signature.ksy (hunk 8 above)
Comment (evdenis) [reply common/samsung_signature.ksy:3]: Only part of the fields reversed. If you know more, please add. Anyway, you also use it https://github.com/kaitai-io/kaitai_struct_formats/pull/302/files#diff-41ce83ce4a8bc1c0b8eb954dc851ef7a877bc4c92b13391a407fe2ffcbab6d2eR33-R34

**Comment by KOLANICH** [comment]
Not quite. In some points your "dupe" is better.

* A better license: CC0 instead of MIT. I have taken MIT to reduce legal risks for me.
* Samsung signature
* Some more fields are documented.

I suggest you to take the parts of my code that are better than yours and add them into yours.

**Pull request closed without merging**